packages/gatsby/cache-dir/socketIo.js

```diff
@@ -88,6 +88,14 @@ export default function socketIo() {
   }
 }
 
+function savePageDataAndStaticQueries(pathname, pageData, staticQueriesData) {
+  pageQueryData[pathname] = pageData
```

Comment:
https://github.com/gatsbyjs/gatsby/blob/a5665d28f585c7f8727e2bbaccb4ab0fffeaa8e7/packages/gatsby/cache-dir/socketIo.js#L57-L62

We do couple things differently when we update `pageQueryData` store due to `pageQueryResult` websocket messages:
 - first, we check if data even changed
 - if data changed - we create copy of `pageQueryData` and "normalize" the key/path:
    - creating new object makes sure that query result store ( https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/cache-dir/query-result-store.js ) component "notices" that and chose wether to re-render (via `shouldComponentUpdate`)
    - normalize just make sure keys are "uniform" (`normalize` just ensure there is no trailing slash pretty much as it works right now)

Reply:
The normalize part isn't really necessary cause I use the pathname from the page-data but I'll add it nonetheless so we're sure there is no friction.

For didDataChange, I'm not sure we need it cause we kinda want to use the latest page-data

Reply:
Yes, but normalized path doesn't actually mean matching path in page-data

As example - if you have `/blog/` page - normalized path will be `/blog`, but page-data would have `/blog/`

The `didDataChange` are checks to see if something changed before we create new clone (and force re-renders in components by creating new object)

Reply:
I see, you're right 👍